### DISCUSSION TITLE
CQI Report on galaxy Samsung S23

### DISCUSSION THREAD

**Discussion by emachona** [christianrowlands/android-network-survey#31] [Q&A] [answered]
Hi Christian,

First of all, thanks for the extensive app, great job!
I am using the app for research purposes, collecting cellular data info. However, I am never getting a value for the CQI parameter in LTE (the csv CQI column is empty). I am logging in North Macedonia, with a Samsung galaxy S23 device (also tried it with Samsung galaxy A71 but the same issue persists). Might this be a bug in the app, or are my devices unsupported for CQI reports? Maybe something else?

Kind regards,
Ema

**Comment by christianrowlands** [comment]
Hey Ema,

Thank you for using the app! It is always great to hear from people using the app.

You are correct that CQI is only supported on certain devices. However, there might be something else to try. On some devices, if the phone is connected to Wi-Fi the CQI value will always be blank. Try turning off Wi-Fi and running the survey again to see if that resolves the problem.

I have a Samsung S22 that I will try to run the app on to see if I can get CQI working on that device.

**Comment by christianrowlands** [comment, marked as answer]
I was able to run NS up on a Galaxy S22 and I was not able to get a valid CQI. The value that the Samsung phone returned for the Android API call ([getCqi](https://developer.android.com/reference/android/telephony/CellSignalStrengthLte#getCqi())) was always `2147483647`, which is defined as [UNAVAILABLE](https://developer.android.com/reference/android/telephony/CellInfo#UNAVAILABLE) in the Android API. 

It seems that Samsung devices don't support CQI in the Android APIs. I have found that Samsung wires up their hardware and other items to the Android APIs in a strange way. FWIW, I typically like running Network Survey on Pixel devices because Google does a better job of supporting all of the Android APIs.

Also, as noted in the other comment I made, other devices (e.g. Pixel 8 Pro) won't have a value for the CQI when connected to Wi-Fi. You must turn off Wi-Fi for CQI to be logged (also true of Timing Advance on certain phones).

**Comment by emachona** [comment]
Thanks for the fast reply. I just got the chance to see it, but sadly I do not own a Pixel to test it out right now. Anyways I am glad to hear that this is not a bug in the code and just a hardware limitation. 
Keep up the good work!
